k8s/src/main/scala/io/buoyant/k8s/Watchable.scala

```diff
@@ -182,6 +184,15 @@ private[k8s] abstract class Watchable[O <: KubeObject: TypeReference, W <: Watch
             AsyncStream.fromFuture(f)
         }
       }
+      // Ignore any cases where we receive a resource version lower
```

Comment:
defining this as a method `def increasingSubsequence[T <: Ordered[T]](stream: AsyncStream[T]): AsyncStream[T]` would make the logic of this method easy to test (since you can just test it on an AsyncStream of numbers).

Reply:
Smart!